nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-security/src/main/java/org/apache/nifi/web/security/oidc/StandardOidcIdentityProvider.java

```diff
@@ -439,8 +439,21 @@ private LoginAuthenticationToken convertOIDCTokenToLoginAuthenticationToken(OIDC
                 identity = claimsSet.getStringClaim(EMAIL_CLAIM);
                 logger.info("The 'email' claim was present. Using that claim to avoid extra remote call");
             } else {
-                identity = retrieveIdentityFromUserInfoEndpoint(oidcTokens);
-                logger.info("Retrieved identity from UserInfo endpoint");
+                final List<String> fallbackClaims = properties.getOidcFallbackClaimsIdentifyingUser();
+                if (fallbackClaims.size() > 0) {
+                    logger.info("fallbackClaims.size() : " + fallbackClaims.size());
+                    for (String fallbackClaim : fallbackClaims) {
+                        if (availableClaims.contains(fallbackClaim)) {
+                            identity = claimsSet.getStringClaim(fallbackClaim);
+                            break;
+                        }
+                    }
+                }
+                if (StringUtils.isBlank(identity)) {
+                    identity = retrieveIdentityFromUserInfoEndpoint(oidcTokens);
+                    logger.info("Retrieved identity from UserInfo endpoint");
```

Comment:
This log message is somewhat unclear and may not be necessary at the info level.  Recommend at least changing it to debug an parameterizing the message with the retrieved identity along the following lines:
```suggestion
                    logger.debug("Identity [{}] retrieved from UserInfo endpoint", identity);
```